### ISSUE TITLE
Feature request: Add buffer at any point in the list (not necessarily append)

### ISSUE THREAD

**Issue by greduan** [rvagg/bl#18]
I have a use-case where I need to be able to put a new buffer in the middle of the BufferList.

Are there any plans to add this, if not, any chances that it'll be added to the plans? :)

If not, any possible alternatives that could offer this use-case?

Thanks. :)

**Comment by mcollina** [comment]
This is an interest use case, but there are no plans to add this functionality (as far as I know, this module is done). I'm :+1: to have this in, if you would like to submit a PR.

I think we might look into add a `splice()` method, just like arrays. What do you think, @rvagg @jcrugzz?

**Comment by rvagg** [comment]
yeah, I like this, but it'll have to come from a PR with some solid tests

**Comment by jcrugzz** [comment]
:+1: This sounds cool but tricky to implement efficiently.

**Comment by greduan** [comment]
Looks like I'll be studying the source code now. :)

If I don't go with another approach to solve the problem, I'll be adding this to `bl`. :)

**Comment by greduan** [comment]
FYI, I won't be implementing this myself. For the moment I no longer have a need for BufferLists like this. Feel free to leave this open for other people to know it has been requested and/or implement it.

**Comment by rvagg** [comment]
Closing as stale
